tutorial/lesson_07_multi_stage_pipelines.cpp

```diff
@@ -30,7 +31,7 @@ int main(int argc, char **argv) {
     // first horizontally, and then vertically.
     {
         // Take a color 8-bit input
-        Image<uint8_t> input = load<uint8_t>("images/rgb.png");
+        Image<uint8_t> input = load<Image<uint8_t>>("images/rgb.png");
```

Comment:
> I really want it to not confuse the audience even for a second

If they look at this source, they're gonna be confused :-)

> How about throwing an exception instead of exiting if WITH_EXCEPTIONS is on?

Rest of code isn't exception-safe, but could be made so with a little effort. Doesn't really help C++-without-exceptions, which is a substantial fraction of C++ usage. Not sure if this is a dealbreaker or not.